tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java

```diff
@@ -63,20 +63,32 @@ private static boolean isNotAdministrativeAgreement(AssociatedArtifact artifact)
         return artifact instanceof File && !(artifact instanceof AdministrativeAgreement);
     }
 
+    private static boolean hasNoFiles(Publication publication) {
+        return publication.getAssociatedArtifacts().stream()
+                   .noneMatch(artifact -> artifact instanceof File);
+    }
+
     private TicketEntry resolvePublishingRequest(TicketEntry ticket, Publication publication,
                                                  PublishingRequestCase publishingRequestCase, Username username)
         throws ApiGatewayException {
         if (REGISTRATOR_PUBLISHES_METADATA_AND_FILES.equals(publishingRequestCase.getWorkflow())) {
-            approveTicket(ticket, username);
             publishPublicationAndFiles(publication);
+            approveTicket(ticket, username);
         }
         if (REGISTRATOR_PUBLISHES_METADATA_ONLY.equals(publishingRequestCase.getWorkflow())) {
-            approveTicket(ticket, username);
```

Comment:
this function should not be there

Reply:
Good. We will check again the bug comming from ticket status for completed Doirequest.